lib/internal/crypto/keys.js

```diff
@@ -328,7 +328,18 @@ function createPublicKey(key) {
   return new PublicKeyObject(handle);
 }
 
-function createPrivateKey(key) {
+function createPrivateKey(key, secret = null) {
+  if (key === 'poly1305') {
+    if (!isArrayBufferView(secret)) {
+      throw new ERR_INVALID_ARG_TYPE(
+        'secret',
+        ['Buffer', 'TypedArray', 'DataView'],
+        secret);
+    }
+    const handle = new KeyObjectHandle(kKeyTypePrivate);
+    handle.initRaw(/* NID_poly1305 */ 1061, secret);
```

Comment:
I think we could export and use `EVP_PKEY_POLY1305` from the binding since it seems like we're already doing that for other types.

Reply:
Agreed, done. This was just the least amount of work to put a proof-of-concept together.